internal/cmd/plugin/logs/pretty/pretty.go

```diff
@@ -0,0 +1,242 @@
+		},
+		Args: cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			bf.loggers = stringset.From(loggers)
+			bf.pods = stringset.From(pods)
+
+			recordChannel := make(chan logRecord)
+			recordGroupsChannel := make(chan []logRecord)
+
+			var wait sync.WaitGroup
+
+			wait.Add(1)
+			go func() {
+				bf.decode(cmd.Context(), os.Stdin, recordChannel)
+				wait.Done()
+			}()
+
+			wait.Add(1)
+			go func() {
+				bf.group(cmd.Context(), recordChannel, recordGroupsChannel)
+				wait.Done()
+			}()
+
+			wait.Add(1)
+			go func() {
+				bf.write(cmd.Context(), recordGroupsChannel, os.Stdout)
+				wait.Done()
+			}()
+
+			wait.Wait()
+			return nil
+		},
+	}
+
+	cmd.Flags().StringSliceVar(&loggers, "loggers", nil,
+		"The list of loggers to receive. Defaults to all.")
+	cmd.Flags().StringSliceVar(&pods, "pods", nil,
+		"The list of pods to receive from. Defaults to all.")
+	cmd.Flags().Var(&bf.minLevel, "min-level",
+		`Hides the messages whose log level is less important than the specified one.
+Should be empty or one of error, warning, info, debug, or trace.`)
+
+	return cmd
+}
+
+// decode progressively decode the logs
+func (bf *prettyCmd) decode(ctx context.Context, reader io.Reader, recordChannel chan<- logRecord) {
+	scanner := bufio.NewScanner(reader)
+
+	for scanner.Scan() {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+		}
+
+		record, err := newLogRecordFromBytes(scanner.Bytes())
+		if err != nil {
+			_, _ = fmt.Fprintln(
+				os.Stderr,
+				aurora.Red(fmt.Sprintf("JSON syntax error (%s)", err.Error())),
+				scanner.Text())
+			continue
+		}
+
+		record.normalize()
+
+		if !bf.isRecordRelevant(record) {
+			continue
+		}
+
+		recordChannel <- *record
+	}
+
+	close(recordChannel)
+}
+
+// group transforms a stream of logs in a stream of log groups
+func (bf *prettyCmd) group(ctx context.Context, logChannel <-chan logRecord, groupChannel chan<- []logRecord) {
+	const bufferMaxEntries = 200
+	var bufferArray [bufferMaxEntries]logRecord
+
+	buffer := bufferArray[0:0]
+
+	pushLogGroup := func() {
+		if len(buffer) == 0 {
+			return
+		}
+
+		bufferCopy := make([]logRecord, len(buffer))
+		copy(bufferCopy, buffer)
+		groupChannel <- bufferCopy
+
+		buffer = bufferArray[0:0]
+	}
+
+logLoop:
+	for {
+		timer := time.NewTimer(1 * time.Second)
+		defer timer.Stop()
+
+		select {
+		case <-ctx.Done():
+			break logLoop
+
+		case <-timer.C:
+			pushLogGroup()
+
+		case logRecord, ok := <-logChannel:
+			if !ok {
+				break logLoop
+			}
+
+			buffer = append(buffer, logRecord)
+			if len(buffer) == bufferMaxEntries {
+				pushLogGroup()
+			}
+		}
+	}
+
+	pushLogGroup()
+	close(groupChannel)
+}
+
+// write write the logs on the output
+func (bf *prettyCmd) write(ctx context.Context, recordGroupChannel <-chan []logRecord, writer io.Writer) {
+	logRecordComparison := func(l1, l2 logRecord) int {
+		if l1.TS < l2.TS {
+			return -1
+		} else if l1.TS > l2.TS {
+			return 1
+		}
```

Comment:
In line with my earlier comment about the grouping of logs into groups to make them sortable:
I have seen some processes producing timestamps in Unix epoch, and some processes producing timestamps in RFC3390 format.
If we had heterogeneous TS types here, this sorting would be unreliable.

Reply:
Which processes are you referring to? I think that with #5783 and https://github.com/cloudnative-pg/machinery/pull/38 we should have all the timestamps in the same format.

If this is not true, we should fix it.

Reply:
Ah, OK, fixed there. That's it, `zap` by default uses RFC3390, so I'd see differences sometimes.